internal/indexcoord/index_builder.go

```diff
@@ -55,8 +55,8 @@ func newIndexBuilder(ctx context.Context, ic *IndexCoord, metaTable *metaTable,
 		meta:             metaTable,
 		ic:               ic,
 		tasks:            make(map[int64]indexTaskState, 1024),
-		notify:           make(chan struct{}, 1024),
-		scheduleDuration: time.Second * 1,
+		notify:           make(chan struct{}, 10),
```

Comment:
change len of notify to 1 and use
select {
   notify <- struct {}{}
  default:
    
}
to notify